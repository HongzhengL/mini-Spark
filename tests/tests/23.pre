python3 tests/23.py > tests/23.rdd