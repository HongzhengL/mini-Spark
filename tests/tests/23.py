import random

random.seed(1234)

class RDD:
    def __init__(self, trans, backed, numpartitions = 8):
        self.backed = backed
        self.numpartitions = numpartitions
        self.trans = trans
        if trans == 0:    
            self.dep0 = None
            self.dep1 = None
        elif trans == 2:
            
            self.dep0 = RDD(3, 0, numpartitions)
            self.dep1 = RDD(3, 0, numpartitions)
        elif trans == 3:
            self.dep0 = RDD(0, random.randint(0,1023), random.randint(1,128))
        else:
            self.dep0 = RDD(0, random.randint(0,1023), numpartitions)
            
    def push(self, rdd):
        if self.trans == 2:
            direction = random.randint(0,1)
            if direction == 0:
                if self.dep0.trans != 0:
                    self.dep0.push(rdd)
                else:
                    self.dep0 = rdd
            else:
                if self.dep1.trans != 0:
                    self.dep1.push(rdd)
                else:
                    self.dep1 = rdd
        elif self.trans != 0:
            if self.trans != 3:
                rdd.numpartitions = self.numpartitions
            if self.dep0.trans != 0:
                self.dep0.push(rdd)
            else:
                self.dep0 = rdd
        else:
            return
    def __str__(self):
        if self.trans == 2:
            return f"2 {self.dep0}{self.dep1}"
        elif self.trans == 3:
            return f"3 {self.numpartitions} {self.dep0}" 
        elif self.trans == 0:
            return f"0 {self.backed} {self.numpartitions} "
        else:
            return f"{self.trans} {self.dep0}"
        
r = RDD(2, 0, 8)
for i in range(3000):
    r.push(RDD(random.randint(0,3), random.randint(0,1023), random.randint(1,64)))
print(r)