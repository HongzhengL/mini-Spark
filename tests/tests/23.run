./tests/22.tmp ./tests/23.rdd