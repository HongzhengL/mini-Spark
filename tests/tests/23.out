1280	4123597
1792	3942931
5248	3763171
128	4135595
7040	3971984
4096	3546728
2944	3722706
7552	3364162
2432	3559645
3584	3670341
3072	3844570
1920	4124612
4224	3869682
4736	3948716
1408	4165467
6528	3875010
6016	4119996
3712	3971685
768	3733001
5888	4157916
3200	3882645
256	3619773
5376	3870157
2560	3927482
7680	3715020
4864	4179152
4352	4473630
5504	4034897
7168	4165790
2048	3869249
7808	4354781
1536	4056259
6656	3889390
1024	3464503
6144	3744243
3840	3991358
896	3618930
384	3819789
512	4077551
2688	4040675
5632	3877213
7296	4154421
0	4113892
2176	3851809
5120	4271890
4992	3916030
4480	3997796
3328	3999876
2816	3958012
7936	3869019
2304	3933860
7424	3714472
6272	4010699
1152	4078169
1664	3983647
6784	3894478
640	4120435
5760	4095022
4608	3826443
3456	4085999
6400	3965613
8064	4252717
3968	4257528
6912	4178116
